desktop/cmp/input/CodeInput.ts

```diff
@@ -326,22 +334,33 @@ class CodeInputModel extends HoistInputModel {
         if (this.cursor) this.clearSearchResults();
     };
 
-    onAutoFormat = () => {
-        if (!isFunction(this.componentProps.formatter)) return;
+    override toInternal(val: any) {
+        return this.componentProps.readonly && this.componentProps.autoFormat
+            ? this.safeFormatValue(val)
+            : val;
+    }
 
-        const editor = this.editor,
-            val = this.tryPrettyPrint(editor.getValue());
-        editor.setValue(val);
-    };
+    private safeFormatValue(val: any) {
+        if (!isFunction(this.componentProps.formatter)) return val;
 
-    tryPrettyPrint(str) {
         try {
-            return this.componentProps.formatter(str);
-        } catch (e) {
-            return str;
+            return this.componentProps.formatter(val);
+        } catch (ignored) {
+            return val;
         }
     }
 
+    private formatValue() {
```

Comment:
@amcclain had some naming / formatting updates he wanted done here.

1. onAutoFormat -> formatValue. We are choosing to use autoFormat as the new prop (on interface) to format readonly data when loading into the UI. Anselm felt onAutoFormat was not the best name of the reaction happening here (to a user requesting the use of the formatter to update their UI)
2.  tryPrettyPrint was use only once, in this method, to run the formatter. Inlining that call to the method itself.